accepted/00000-ansible-integration.md

```diff
@@ -0,0 +1,327 @@
+
+In short, we would gather, for instance:
+- Ansible inventory (hosts)
+- Ansible SSH keys referenced in the inventory
+- Playbooks under specified "remote playbook path" (to be displayed in the UI - readonly mode - not editable)
+
+This would be stored under `/var/lib/spacewalk/ansible/inventory-label/` path in the Uyuni server. At the time of processing the inventory file by "ansible-gatherer", it needs to be tailored to adapt the path of the SSH keys to the local path in the Uyuni server. As done in script from Ansible Integration PoC [here](https://github.com/meaksh/uyuni-hacks/blob/master/scripts/ansible/import_systems_from_ansible_controller.py).
+
+With this information, Java is able to proceed creating these systems in Uyuni, as "Foreign / ANSIBLE" system type.
+
+- If any of those systems is already existing with "ANSIBLE" system type, then nothing to do
+- If the system is already registered but not with "ANSIBLE" system type, then add "ANSIBLE" system type.
+
+Of course, we need some new tables in the database, at least something like:
+
+```
+"suseAnsibleController" (label, type, org_1)
+"suseAnsibleControllerConfig" (control_node_id, parameter, value)
+"suseAnsibleControllerSystem" (system_id, control_node_id)
+"suseAnsibleControllerPlaybook" (control_node_id, local_path, content?)
+```
+
+And also create the new "ANSIBLE" system type, which should be an addon-system type compatible with Salt, Foreign and, in case this doesn't need any extra effors, maybe also for a potential "Management/ANSIBLE" (traditional client) system type. Here is the [SQL script](https://github.com/meaksh/uyuni-hacks/blob/master/scripts/ansible/add_ansible_entitleme.sql) used for PoC demo. It's also needed to adapt the rhn databaste functions implemented in the DB.
+
+NOTE: The profile for a system with registered in Uyuni as "Foreign/ANSIBLE" does not look like as the one Salt/Traditional system, since most of the Uyuni features (channels, software profile, etc) are not implemented for neither "Foreign" nor "ANSIBLE" system types. In order to get the entire portfolio of Uyuni features for Salt minion, this "Foreign/ANSIBLE" systems would need to be transitioned to "Salt/ANSIBLE" minion.
+
+Notice also that the Ansible control node is not listed as systems list page, since there is not system entry for it, it would be shown in the respective new UI page for managing Ansible control nodes.
+
+#### New UI tabs & pages
+
+* Ansible
+* Ansible / Ansible control nodes list
+* Ansible / Add new Ansible control node
+* Ansible / View Ansible control node information & and visualize playbooks (would also allow triggering playbooks via "salt-ssh" in the control node)
+* Systems / Systems list / Ansible (show only Ansible systems and its control node)
+
+or even create a new entry level on the menu as "Automation", something like:
+
+* Automation / Ansible / ...
+* Automation / Salt / ...
+* Automation / Remote commands
```

Comment:
This is new and surprising. What should it do? Is this related to Ansible somehow? Or would it be better put in a separate RFC?

Reply:
No, it's not. It's a UI thing that I mentioned a month ago, on how to display this:

https://github.com/uyuni-project/uyuni-rfc/pull/53#discussion_r583784531

![image](https://user-images.githubusercontent.com/50680111/112738100-27112700-8f60-11eb-8494-2ee7c994cafc.png)

But we'll need @ncounter / @cbbayburt to provide insight

Reply:
I gave an overview of my thinking in https://github.com/uyuni-project/uyuni-rfc/pull/53#discussion_r602851071 . Actually I'd stay attached to the approach of splitting first the "managed VS not managed" criteria, then I'd wrap whatever is meant to wrap.
If for some reason this concept cannot be respected, I could be in favor of this "Automation" wrapping idea. But the main aspect I am not comfortable with is to show and present "Ansible" at the very same level of "Salt", which we know it is definitely not from the Uyuni perspective. Salt is the engine and the core of the configuration management Uyuni is built on, Ansible is just an external type we let it in but we don't touch at all instead. And I also wonder what if yet-another-type will come in in the future.